### PR TITLE
Add - tests for service layer

### PR DESCRIPTION
**Pull request by ChristofferGamel**
# Description
## What is added/changed/removed
We added phpunit tests for the construction of a livequiz

## Issue/fixes reference
Added tests for the service layer of: #37 

## Testing
Run the phpunit tests :) 

# Checks
- [x] Updated / added tests for these changes
- [x] PHPunit locally passed
- [x] Codesniffer locally passed
- [x] Behat locally passed

### CODE REVIEW

**Review by Laczse — APPROVED** [review]
Looks good

**Comment by DeeKahy** [comment]
The branch isn't up to date anymore. Also I'd like behat to be merged before we merge this one #67

**Review by ZeroConfidence — DISMISSED** [review]
Looks good

File: server/moodle/mod/livequiz/classes/models/livequiz.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ class livequiz {
      * @param int $timecreated
      * @param int $timemodified
      */
-    private function __construct(
+    public function __construct(
```
Comment (Laczse) [inline server/moodle/mod/livequiz/classes/models/livequiz.php:84]: Hvorfor er denne lavet public i stedet for private? Vi skal vel aldrig selv oprette et livequiz instance?

**Review by Laczse — CHANGES_REQUESTED** [review]

File: server/moodle/mod/livequiz/classes/models/livequiz.php (hunk 1 above)
Comment (DeeKahy) [reply server/moodle/mod/livequiz/classes/models/livequiz.php:84]: i just tried running it, and it works when you change it to private. So there probably is a good reason @ChristofferGamel changed everything to public. (yes i used git blame 😎 )

File: server/moodle/mod/livequiz/tests/phpunit/livequiz_service_test.php [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * LiveQuiz Service Test Class
+ *
+ * This class contains unit tests for the LiveQuiz service functionality.
+ *
+ * @package mod_livequiz
+ * @copyright 2023
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
```
Comment (DeeKahy) [inline server/moodle/mod/livequiz/tests/phpunit/livequiz_service_test.php:24]: where did this license come from?

**Review by DeeKahy — DISMISSED** [review]
Great job! I still want to answer for the public instead of private thing.

File: server/moodle/mod/livequiz/tests/phpunit/livequiz_service_test.php (hunk 2 above)
Comment (ChristofferGamel) [reply server/moodle/mod/livequiz/tests/phpunit/livequiz_service_test.php:24]: Same license used in other files. Is it not the right one?

https://github.com/AAU-P5-Moodle/moodle-1/blob/main/server/moodle/mod/livequiz/view.php
view.php for instance:

image

Do you want it updated to:

```
 * LiveQuiz Service Test Class
 *
 * This class contains unit tests for the LiveQuiz service functionality.
 *
 * @package   mod_livequiz
 * @copyright 2024 Software AAU
 * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
 */
```

File: server/moodle/mod/livequiz/classes/models/livequiz.php (hunk 1 above)
Comment (ChristofferGamel) [reply server/moodle/mod/livequiz/classes/models/livequiz.php]: This wasnt the intention. Good observation 👍

**Review by DeeKahy — APPROVED** [review]
am doing it agiain

**Pull request merged**